org.eclipse.winery.frontends/app/tosca-management/src/app/wineryLicenseModule/wineryLicense.component.ts

```diff
@@ -11,84 +11,284 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
  *******************************************************************************/
-import { Component, OnInit } from '@angular/core';
+import { ViewChild, Component, OnInit, TemplateRef } from '@angular/core';
 import { WineryLicenseService } from './wineryLicense.service';
 import { WineryNotificationService } from '../wineryNotificationModule/wineryNotification.service';
 import { InstanceService } from '../instance/instance.service';
 import { ToscaTypes } from '../model/enums';
 import { HttpErrorResponse } from '@angular/common/http';
 import { catchError, map } from 'rxjs/internal/operators';
-import { forkJoin, throwError } from 'rxjs/index';
+import { forkJoin } from 'rxjs';
 import { Observable } from 'rxjs/Rx';
+import {
+    WineryRepositoryConfigurationService
+} from '../wineryFeatureToggleModule/WineryRepositoryConfiguration.service';
+import { saveAs } from 'file-saver';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MatStepper } from '@angular/material/stepper';
+import { BsModalRef, BsModalService } from 'ngx-bootstrap';
+import { LicenseEngineService } from './licenseEngine.service';
 
 @Component({
     templateUrl: 'wineryLicense.component.html',
     styleUrls: ['wineryLicense.component.css'],
-    providers: [WineryLicenseService]
+    providers: [WineryLicenseService, LicenseEngineService]
 })
 
 export class WineryLicenseComponent implements OnInit {
 
-    licenseText = '';
-    initialLicenseText = '';
-    licenseAvailable = true;
-    licenseType = '';
-
     loading = true;
-    options: any;
+    loadingLicense = false;
     isEditable = false;
+    licenseAvailable = true;
+    showForm = false;
+    loadingbar = false;
+
+    foundLicenses = '';
+    currentLicenseText = '';
+    initialLicenseText = '';
+    selectedLicenseText = '';
+    licenseType = '';
+    options: string[] = [];
+    selectedOptions: string[] = [];
+    compatibleLicenses: string[] = [];
 
     toscaType: ToscaTypes;
+    licenseEngine: boolean;
+    firstFormGroup: FormGroup;
+    secondFormGroup: FormGroup;
+    confirmSaveModalRef: BsModalRef;
+    confirmDownloadModalRef: BsModalRef;
+
+    @ViewChild('stepper') stepper: MatStepper;
+    @ViewChild('confirmSaveModal') confirmSaveModal: TemplateRef<any>;
+    @ViewChild('confirmDownloadModal') confirmDownloadModal: TemplateRef<any>;
 
-    constructor(private service: WineryLicenseService,
-                private notify: WineryNotificationService,
-                public sharedData: InstanceService) {
-        this.toscaType = this.sharedData.toscaComponent.toscaType;
```

Comment:
What about the toscaType? Where is it set?

Reply:
fixed; line was missing